metron-platform/metron-elasticsearch/README.md

```diff
@@ -33,7 +42,217 @@ For instance, an `es.date.format` of `yyyy.MM.dd.HH` would have the consequence
+	</td>
+</tr>
+<tr>
+	<td>
+	yes
+	</td>
+	<td>
+<pre><code>"my_property": {
+  "type": "string",
+  "index": "analyzed"
+}
+</code></pre>
+	</td>
+	<td>
+<pre><code>"my_property": {
+  "type": "text",
+  "fields": {
+    "keyword": {
+      "type": "keyword",
+      "ignore_above": 256
+    }
+  }
+}
+</code></pre>
+	</td>
+	<td>
+	"New York" searchable as single value or as text document, can aggregate and sort on the sub term "keyword."
+	</td>
+</tr>
+</table>
+
+If you want to set default string behavior for all strings for a given index and type, you can do so with a mapping similar to the following (replace ${your_type_here} accordingly):
+
+```
+# curl -XPUT 'http://${ES_HOST}:${ES_PORT}/_template/default_string_template' -d '
+{
+  "template": "*",
+  "mappings" : {
+    "${your_type_here}": {
+      "dynamic_templates": [
+        {
+          "strings": {
+            "match_mapping_type": "string",
+            "mapping": {
+              "type": "text"
+            }
+          }
+        }
+      ]
+    }
+  }
+}
+'
+```
+
+By specifying the "template" property with value "*" the template will apply to all indexes that have documents indexed of the specified type (${your_type_here}). This results in the following template.
+
+```
+# curl -XGET 'http://${ES_HOST}:${ES_PORT}/_template/default_string_template?pretty'
+{
+  "default_string_template" : {
+    "order" : 0,
+    "template" : "*",
+    "settings" : { },
+    "mappings" : {
+      "${your_type_here}" : {
+        "dynamic_templates" : [
+          {
+            "strings" : {
+              "match_mapping_type" : "string",
+              "mapping" : {
+                "type" : "text"
+              }
+            }
+          }
+        ]
+      }
+    },
+    "aliases" : { }
+  }
+}
+```
+
+Notes on other settings for types in ES
+* doc_values
+    * on-disk data structure
+    * provides access for sorting, aggregation, and field values
+    * stores same values as _source, but in column-oriented fashion better for sorting and aggregating
+    * not supported on text fields
+    * enabled by default
+* fielddata
+    * in-memory data structure
+    * provides access for sorting, aggregation, and field values
+    * primarily for text fields
+    * disabled by default because the heap space required can be large
+
+
+##### Type Mapping References
+* [https://www.elastic.co/guide/en/elasticsearch/reference/5.6/mapping.html](https://www.elastic.co/guide/en/elasticsearch/reference/5.6/mapping.html)
+* [https://www.elastic.co/guide/en/elasticsearch/reference/5.6/breaking_50_mapping_changes.html](https://www.elastic.co/guide/en/elasticsearch/reference/5.6/breaking_50_mapping_changes.html)
+* [https://www.elastic.co/blog/strings-are-dead-long-live-strings](https://www.elastic.co/blog/strings-are-dead-long-live-strings)
+
+## Using Metron with Elasticsearch 5.6.2
```

Comment:
The ES 2.x nested alert field still exists in this branch, right?  I don't think that requirement got removed.

If not, this section should probably just say it's required in ES, and drop the obsolete portion of the statement..

Reply:
All set